### ISSUE TITLE
Feature Request: `ASSET_HOST` to enhance static asset delivery

### ISSUE THREAD

**Issue by dmeremyanin** [metabase/metabase#49458]
**Is your feature request related to a problem? Please describe.**

Our team is distributed across the globe, and the asset loading times can be slow for users located far from the application host. This can lead to a less-than-ideal user experience.

**Describe the solution you'd like**

An `MB_ASSET_HOST` configuration option would be highly beneficial, as it would allow to specify an alternative host for serving static assets. This would enable the use of a CDN or a dedicated asset server.

**Describe alternatives you've considered**

We have considered configuring a reverse proxy to replace all asset URLs using [sub_filter](https://nginx.org/en/docs/http/ngx_http_sub_module.html#sub_filter) in nginx to serve static assets from a different location. However, this method is complex and requires ongoing maintenance. A built-in configuration option would greatly simplify the process.

**How important is this feature to you?**

This feature would be a valuable enhancement, as it could improve performance and user experience in Metabase for teams like ours that are distributed across the globe.

**Additional context**

Implementing this feature would align with best practices for web performance optimization, as it's not always possible to use the same domain for dynamic and static content.

**Comment by brunobergher** [comment]
@dmeremyanin we were discussing this internally just the other day. One thing which has discouraged us from moving forward though is that, in general, DB latency is a lot bigger than asset download latency. So while this would make things better, it would probably not make a big impact.

Have you considered this angle? How would you handle DB access latency for your users?

**Comment by dmeremyanin** [comment]
I agree that DB latency is a more significant factor overall, but as @paoliniluis mentioned, the size of Metabase's static assets could still be a reason to consider this change. For instance, Metabase loads around 3MB of Brotli-compressed assets on the main page, which makes up about 90% of all the transferred traffic (3.3MB in total).

Using a CDN can improve responsiveness, and it's a common best practice in web development to offload static content. A good example of this approach can be seen in the [Ruby on Rails asset pipeline](https://guides.rubyonrails.org/asset_pipeline.html#set-up-a-cdn-to-serve-static-assets), where static assets are configured via `config.asset_host = "assets.metabase-is-awesome.com"`.

Thanks for considering this, and I'd be happy to discuss further or clarify any details.

<img width="1371" alt="Image" src="https://github.com/user-attachments/assets/61e09005-3e7d-4d7c-bae1-f952695711ce" />

**Comment by brunobergher** [comment]
Completely agree, @dmeremyanin. I'll work on prioritizing this on our side. Thank you.
